### PR TITLE
Add frontend tests

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

File: frontend/src/js/__tests__/Common/ConfirmationContent.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const {asFragment} = render(
+      <ConfirmationContent
+        data={minProps.data}
+        channel={{}}
+      />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+  it('should call delete handler function on yes confirmation', () => {
+    ApplicationsStore.deleteApplication = mockResolver;
+    const deleteApps = ApplicationsStore.deleteApplications;
```
Comment (joaquimrocha) [inline frontend/src/js/__tests__/Common/ConfirmationContent.spec.js:27]: I think you meant `deleteApplication` (singular), and this call should be called before overriding the function with the `mockResolver` (because AFAIU, this is about restoring the overridden function to the original one, right?).

File: frontend/src/js/__tests__/Common/ModalButton.spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import {fireEvent, getAllByTestId, render} from '@testing-library/react';
+import React from 'react';
+import ModalButton from '../../components/Common/ModalButton.react';
+import { theme } from '../../TestHelpers/theme.js';
+
+describe('Modal Button', () => {
+  it('should render Application Edit Dialog on Add Icon click', () => {
+    const {getByTestId} = render(
+      <ModalButton data={{}}
+        modalToOpen = "AddApplicationModal"
```
Comment (joaquimrocha) [inline frontend/src/js/__tests__/Common/ModalButton.spec.js:11]: nitpick: We don't use spaces around `=` in element properties.

File: frontend/src/js/__tests__/Common/ModalButton.spec.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      />);
+    fireEvent.click(getByTestId('modal-button'));
+    expect(getByTestId('app-edit-form')).toBeInTheDocument();
+  });
+  it('should render AddGroupModal on Add Icon click', () => {
+    const {getByTestId} = render(
+      <ModalButton
+        data={{}}
+        modalToOpen = "AddGroupModal"
```
Comment (joaquimrocha) [inline frontend/src/js/__tests__/Common/ModalButton.spec.js:20]: nitpick: We don't use spaces around `=` in element properties.

File: frontend/src/js/__tests__/Common/ModalButton.spec.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      />);
+    fireEvent.click(getByTestId('modal-button'));
+    expect(getByTestId('group-edit-form')).toBeInTheDocument();
+  });
+  it('should render AddChannelModal on Add Icon click', () => {
+    const tree = (
+      <MuiThemeProvider theme={theme}>
+        <ModalButton data={{}}
+          modalToOpen = "AddChannelModal"
```
Comment (joaquimrocha) [inline frontend/src/js/__tests__/Common/ModalButton.spec.js:29]: nitpick: We don't use spaces around `=` in element properties.

File: frontend/src/js/__tests__/Common/ModalButton.spec.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        />
+      </MuiThemeProvider>
+    );
+    const {getByTestId} = render(
+            tree);
```
Comment (joaquimrocha) [inline frontend/src/js/__tests__/Common/ModalButton.spec.js:34]: Please don't wrap for a one small word parameter.

File: frontend/src/js/components/Applications/EditDialog.js [hunk 6]
```diff
@@ -78,7 +78,6 @@ function EditDialog(props) {
               helperText="Clone channels and groups from another other application"
               margin="normal"
               component={TextField}
-              displayEmpty
```
Comment (joaquimrocha) [inline frontend/src/js/components/Applications/EditDialog.js:81]: Why is this removed here? Is it an error?

File: frontend/src/js/__tests__/Common/ConfirmationContent.spec.js (hunk 1 above)
Comment (ashu8912) [reply frontend/src/js/__tests__/Common/ConfirmationContent.spec.js:27]: yes its like mocking the http function and we check whether that function was called or not

File: frontend/src/js/components/Applications/EditDialog.js (hunk 6 above)
Comment (ashu8912) [reply frontend/src/js/components/Applications/EditDialog.js:81]: Yes it was there for no reason and was also throwing warning about this prop

File: frontend/src/js/__tests__/Common/ModalButton.spec.js (hunk 4 above)
Comment (ashu8912) [reply frontend/src/js/__tests__/Common/ModalButton.spec.js]: done

File: frontend/src/js/__tests__/Common/ModalButton.spec.js (hunk 3 above)
Comment (ashu8912) [reply frontend/src/js/__tests__/Common/ModalButton.spec.js]: done

File: frontend/src/js/__tests__/Common/ModalButton.spec.js (hunk 2 above)
Comment (ashu8912) [reply frontend/src/js/__tests__/Common/ModalButton.spec.js]: done

File: frontend/src/js/__tests__/Common/ConfirmationContent.spec.js (hunk 1 above)
Comment (ashu8912) [reply frontend/src/js/__tests__/Common/ConfirmationContent.spec.js]: done
